pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SparkRBackend.scala

```diff
@@ -80,18 +78,28 @@ class SparkRBackend {
 object SparkRBackend {
   def main(args: Array[String]) {
     if (args.length < 1) {
-      System.err.println("Usage: SparkRBackend <port>")
+      System.err.println("Usage: SparkRBackend <path>")
       System.exit(-1)
     }
     val sparkRBackend = new SparkRBackend()
     try {
-      sparkRBackend.init(args(0).toInt)
+      // bind to random port
+      val boundPort = sparkRBackend.init()
+      // tell the R process via temporary file
+      val path = args(0)
+      val f = new File(path + ".tmp")
+      val output = new FileOutputStream(f)
```

Comment:
Can combine this into one line `val dos = new DataOutputStream(FileOutputStream(f))` ?